### DISCUSSION TITLE
How do I use getifaddr() in Gramine Shielded Containers?

### DISCUSSION THREAD

**Discussion by fwoodruff** [gramineproject/gramine#1797] [Q&A]
I have a (third-party) C++ program I am able to create a Docker image for (built with Make). I have tried building a 'Graminised' version to run inside SGX. The program produces the same logs initially but differs after:

```
getNetworkInfo() failed calling getifaddrs() iResult=-1=Unknown error -1
```

I believe [this](https://man7.org/linux/man-pages/man3/getifaddrs.3.html) is the Linux library function I need supported. Is this something I can achieve by updating the Gramine manifest file for example?

**Comment by dimakuv** [comment]
`getifaddrs()` uses the AF_NETLINK socket type which is **not** implemented in Gramine, at least yet.

You can find some info and potential workarounds here:
- https://github.com/gramineproject/gramine/issues/948
